plugin/sampling/strategystore/adaptive/aggregator.go

```diff
@@ -112,9 +129,23 @@ func (a *aggregator) RecordThroughput(service, operation string, samplerType spa
 
 func (a *aggregator) Start() {
 	go a.runAggregationLoop()
+	a.postAggregator.Start()
 }
 
 func (a *aggregator) Close() error {
-	close(a.stop)
+	a.Lock()
+	defer a.Unlock()
+
+	if err := a.postAggregator.Close(); err != nil {
```

Comment:
May not need separate close if we combine bg loops, but otherwise you cannot exit on error here, need to make sure the other loop is stopped too. Normally we combine errors to a slice and return via errors.Join()

Reply:
done